src/nvim/plines.c

```diff
@@ -189,10 +189,11 @@ int plines_win_col(win_T *wp, linenr_T lnum, long column)
 /// @param[out] nextp    if not NULL, the line after a fold
 /// @param[out] foldedp  if not NULL, whether lnum is on a fold
 /// @param[in]  cache    whether to use the window's cache for folds
+/// @param[in]  winheight when true limit to window height
```

Comment:
Added a `winheight` argument to pass down to `plines_win_nofill()`. This lets us re-use `plines_win_full()` in the newly added `plines_correct_topline()` that needs to first `adjust_plines_for_skipcol()` before limiting to window height itself.